### PR TITLE
feat(license): add new flag for classifier confidence level

### LINKED ISSUE #3378: Config flag for `confidence level` in the license scanner
## Description

The [SPDX license list data](https://github.com/spdx/license-list-data/tree/main/text) contains 553 license files, but trivy detects 227 licenses running:

```bash
trivy fs --security-checks license --license-full <path/to/spdx/list/data> -o license.log
```

See output file [license.log](https://github.com/aquasecurity/trivy/files/10345220/license.log).

## What did you expect to happen?

The confidence level in [classifier.go](https://github.com/aquasecurity/trivy/blob/5190f9566b6ad5c0cf914870732891509541369a/pkg/licensing/classifier.go#L57) is set to 0.9. With a configurable confidence level I expect a much better result.

## Output of `trivy -v`:

```
Version: 0.35.0
```

### PR DESCRIPTION
**Pull request by thevibegod**
## Description
- Add new flag `--license-confidence-level` that accepts a float value(default 0.9)
- Make use of the above flag's value in `Classify` method of `classifier.go`.

## Related issues
- Close #3378 

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [x] I've added usage information (if the PR introduces new options)

### CODE REVIEW

**Review by DmitriyLewen — COMMENTED** [review]
Hello @thevibegod 
Thanks for your work!

Resolve conflicts(pull of main branch should help), please.

I added some comments. Take a look, please.

Also i think we need to add more information about confidence level to [docs](https://github.com/aquasecurity/trivy/blob/main/docs/docs/scanner/license.md). (default value, how value affects to number of licenses, etc...)

File: docs/docs/references/cli/trivy_kubernetes.md [hunk 1]
```diff
@@ -27,64 +27,65 @@ trivy kubernetes [flags] { cluster | all | specific resources like kubectl. eg:
...
+      --ignore-policy string             specify the Rego file path to evaluate each vulnerability
+      --ignore-unfixed                   display only fixed vulnerabilities
+      --ignored-licenses strings         specify a list of license to ignore
+      --ignorefile string                specify .trivyignore file (default ".trivyignore")
+      --include-non-failures             include successes and exceptions, available with '--scanners config'
+      --java-db-repository string        OCI repository to retrieve trivy-java-db from (default "ghcr.io/aquasecurity/trivy-java-db")
+      --k8s-version string               specify k8s version to validate outdated api by it (example: 1.21.0)
+      --kubeconfig string                specify the kubeconfig file path to use
+      --license-confidence-level float   specify classifier confidence level (default 0.9)
+      --license-full                     eagerly look for licenses in source code headers and license files
```
[the remark this answers is not in the document]
Comment (thevibegod) [reply docs/docs/references/cli/trivy_kubernetes.md]: Made the above mentioned changes.

File: docs/docs/references/cli/trivy_vm.md [hunk 2]
```diff
@@ -20,59 +20,60 @@ trivy vm [flags] VM_IMAGE
...
+      --helm-values strings              specify paths to override the Helm values.yaml files
+  -h, --help                             help for vm
+      --ignore-policy string             specify the Rego file path to evaluate each vulnerability
+      --ignore-unfixed                   display only fixed vulnerabilities
+      --ignored-licenses strings         specify a list of license to ignore
+      --ignorefile string                specify .trivyignore file (default ".trivyignore")
+      --include-non-failures             include successes and exceptions, available with '--scanners config'
+      --java-db-repository string        OCI repository to retrieve trivy-java-db from (default "ghcr.io/aquasecurity/trivy-java-db")
+      --license-confidence-level float   specify classifier confidence level (default 0.9)
+      --license-full                     eagerly look for licenses in source code headers and license files
```
[the remark this answers is not in the document]
Comment (thevibegod) [reply docs/docs/references/cli/trivy_vm.md]: Made the above mentioned changes.

File: pkg/fanal/analyzer/language/golang/mod/mod.go [hunk 3]
```diff
@@ -50,13 +50,16 @@ type gomodAnalyzer struct {
 
 	// go.mod/go.sum in dependencies
 	leafModParser godeptypes.Parser
+
+	opt analyzer.AnalyzerOptions
```
[the remark this answers is not in the document]
Comment (thevibegod) [reply pkg/fanal/analyzer/language/golang/mod/mod.go]: Made the above mentioned changes.

File: pkg/fanal/analyzer/analyzer.go [hunk 4]
```diff
@@ -143,8 +144,9 @@ type PostAnalysisInput struct {
 }
 
 type AnalysisOptions struct {
-	Offline      bool
-	FileChecksum bool
+	Offline                   bool
+	FileChecksum              bool
+	ClassifierConfidenceLevel float64
```
[the remark this answers is not in the document]
Comment (thevibegod) [reply pkg/fanal/analyzer/analyzer.go]: Made the above mentioned changes.

**Comment by thevibegod** [comment]
@DmitriyLewen Thanks for reviewing and providing the suggestions. Have made all the above mentioned changes and also updated the docs with examples. Please review them and let me know if there are any more changes to be done.

**Review by DmitriyLewen — APPROVED** [review]
Thanks for your work!
LGTM.

**Comment by DmitriyLewen** [comment]
@thevibegod can you run `mage docs:generate` to update CLI docs and fix test?

**Comment by thevibegod** [comment]
> @thevibegod can you run `mage docs:generate` to update CLI docs and fix test?

@DmitriyLewen I have updated the docs.

**Comment by DmitriyLewen** [comment]
cool! thanks!

**Comment by knqyf263** [comment]
Thanks!

**Pull request merged**